types/instance.d.ts

```diff
@@ -224,14 +224,14 @@ export interface FastifyInstance<
    * Set the 404 handler
    */
   setNotFoundHandler<RouteGeneric extends RouteGenericInterface = RouteGenericInterface>(
-    handler: (request: FastifyRequest<RouteGeneric, RawServer, RawRequest>, reply: FastifyReply<RawServer, RawRequest, RawReply, RouteGeneric>) => void
+    handler: (request: FastifyRequest<RouteGeneric, RawServer, RawRequest>, reply: FastifyReply<RawServer, RawRequest, RawReply, RouteGeneric>) => void | Promise<any>
```

Comment:
This method doesn't return anything:

https://github.com/fastify/fastify/blob/21578f56dea6bfd3de1c9a4df497fd535d174221/fastify.js#L499-L503

I think it is a bug in the core!

Reply:
If `setNotFoundHandler` accepts async functions then handler can return `Promise`

Reply:
If `setErrorHandler` returns this then maybe `setNotFoundHandler` should also return the instance?

Reply:
The type changed here is the return type of the `handler` argument to `setNotFoundHandler`, not the return type of `setNotFoundHandler` (which is still `void` and not changed by this PR). The handler is handled by the same response-generating code as normal route handlers:

https://github.com/fastify/fastify/blob/f4cb706144bbf2ebb67257914f78020126488ba5/lib/reply.js#L639

Reply:
@ethanwu10 yes. Here we just found that there is a problem also in setNotFoundHanlder. It is not related to this PR though. I am still not sure about returning any vs unknown in the handler. What do you think?

Reply:
Hmm, the return types of the `set*Handler` methods does seem to be a bug. `setErrorHandler` returns the instance, but is typed as returning `void`, and `setNotFoundHandler` should probably also return the instance as @fox1t mentioned.

Reply:
I think it is better to open another issue to resolve it. Would you mind to open it?